### PR TITLE
Use mamba to spin up development environments

### PR DESCRIPTION
**Pull request by k-dominik**
This updates the contributing instructions with use of mamba instead of conda for envrionment creation. Also update `devenv.py` accordingly.

### CODE REVIEW

**Comment by k-dominik** [comment]
what do you think @emilmelnikov, should we already advocate [mambaforge](https://github.com/conda-forge/miniforge#mambaforge) instead of miniforge?

**Comment by emilmelnikov** [comment]
> what do you think @emilmelnikov, should we already advocate [mambaforge](https://github.com/conda-forge/miniforge#mambaforge) instead of miniforge?

Yes, definitely, and also tell people to `channel_priority: strict` and `channels: [conda-forge]` to our docs in case if someone still wants/needs to use the vanilla conda.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,24 +29,28 @@ The following text equips you with knowledge that makes contributing to ilastik
...
    ```
 
-1. Configure conda to use [strict channel priority](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority):
+1. We recommend to downloading/using Mambaforge to interact with conda.
+   Mambaforge already comes configured correctly (this step can be skipped in this case).
+   If you are using [Miniconda][miniconda] instead, please make sure to configure conda to use [strict channel priority][strict-channel-priority], and install the fast [Mamba][mamba] solver for conda environments:
 
    ```
    conda config --set channel_priority strict
+   conda install --name-base -c conda-forge mamba
```
Comment (emilmelnikov) [inline CONTRIBUTING.md:38]:
```suggestion
   conda install --name base -c conda-forge mamba
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,24 +29,28 @@ The following text equips you with knowledge that makes contributing to ilastik
...
    ```
 
 1. Install [conda-build][conda-build] in order to access the [conda develop][conda-develop] command:
 
    ```
-   conda install --name base -c conda-forge conda-build
+   mamba install --name base -c conda-forge conda-build
```
Comment (emilmelnikov) [inline CONTRIBUTING.md:44]: Are we also moving to `boa` from `conda-build`?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,24 +29,28 @@ The following text equips you with knowledge that makes contributing to ilastik
    gh repo clone ilastik
    ```
 
-1. Configure conda to use [strict channel priority](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority):
+1. We recommend to downloading/using Mambaforge to interact with conda.
+   Mambaforge already comes configured correctly (this step can be skipped in this case).
+   If you are using [Miniconda][miniconda] instead, please make sure to configure conda to use [strict channel priority][strict-channel-priority], and install the fast [Mamba][mamba] solver for conda environments:
```
Comment (emilmelnikov) [inline CONTRIBUTING.md:34]:
```suggestion
1. If you have an existing [Miniconda][miniconda] installation, you can use it for ilastik, but make sure to install [mamba][mamba] and configure conda to use [strict channel priority][strict-channel-priority]:
```

**Review by emilmelnikov — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (k-dominik) [reply CONTRIBUTING.md:44]: when using conda-develop it doesn't make a difference. but yeah, for actual building packages I'm using boa right now.

**Pull request merged**